pennylane/collections/sum.py

```diff
@@ -48,6 +50,8 @@ def sum(x):
     >>> cost(x)
     tensor(0.9177, dtype=torch.float64)
     """
+    warnings.warn("The sum function is deprecated and it will be removed soon.", UserWarning)
```

Comment:
Same thing as @mudit2812 said for `dot` - if we are keeping `qml.sum`, it will just be `qml.op_sum` now, I think the deprecation message would be clearer if it indicated that its specifically `collections.sum` that is being deprecated.

Reply:
Nice call! I added of the collections module to make reference to this specific function.

I don't want to add that qml.sum will redirect to pennylane/ops/functions/dot.py to avoid making the error message too large.